src/test/java/com/nashtech/controller/ReactiveDataControllerTest.java

```diff
@@ -0,0 +1,122 @@
+    }
+
+    @Test
+    void testGetAllBrands_ReactiveDataServiceReturnsError() {
+        // Setup
+        when(reactiveDataService.getAllBrands()).thenReturn(Flux.error(new Exception("message")));
+
+        // Run the test
+        final Flux<CarBrand> result = reactiveDataController.getAllBrands();
+
+        // Verify the results
+        StepVerifier.create(result)
+                .expectErrorMatches(throwable -> throwable instanceof Exception && throwable.getMessage().equals("message"))
+                .verify();
+    }
+
+    @Test
+    void testGetAllBrands_ReactiveDataServiceReturnsNoItem() {
+        // Setup
+        when(reactiveDataService.getAllBrands()).thenReturn(Flux.empty());
+
+        // Run the test
+        final Flux<CarBrand> result = reactiveDataController.getAllBrands();
+
+        // Verify the results
+        StepVerifier.create(result)
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+    @Test
+    void testGetCarsByBrand_ReactiveDataServiceReturnsError() {
+        // Setup
+        when(reactiveDataService.getCarsByBrand("brand")).thenReturn(Flux.error(new Exception("message")));
+
+        // Run the test
+        final Flux<Car> result = reactiveDataController.getCarsByBrand("brand");
+
+        // Verify the results
+        StepVerifier.create(result)
+                .expectErrorMatches(throwable -> throwable instanceof Exception && throwable.getMessage().equals("message"))
+                .verify();
+    }
+
+    @Test
+    void testGetCarsByBrand_ReactiveDataServiceReturnsNoItem() {
+        // Setup
+        when(reactiveDataService.getCarsByBrand("brand")).thenReturn(Flux.empty());
+
+        // Run the test
+        final Flux<Car> result = reactiveDataController.getCarsByBrand("brand");
+
+        // Verify the results
+        StepVerifier.create(result)
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+
+    @Test
+    void testPushDataToCloud() {
+        ResponseEntity<Object> response = reactiveDataController.pushDataToCloud();
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+        verify(reactiveDataService).fetchAndSendData();
+    }
+
+    @Test
+    void testGetCarsByBrand() {
+        String brand = "Toyota";
+        when(reactiveDataService.getCarsByBrand(brand)).thenReturn(Flux.just(new Car(), new Car()));
+
+        Flux<Car> carsFlux = reactiveDataController.getCarsByBrand(brand);
+
+        StepVerifier.create(carsFlux)
+                .expectNextCount(2)
+                .verifyComplete();
+    }
+
+    @Test
+    void testGetAllBrands() {
+        when(reactiveDataService.getAllBrands()).thenReturn(Flux.just(new CarBrand("Toyota"), new CarBrand("BMW")));
+
+        Flux<CarBrand> brandsFlux = reactiveDataController.getAllBrands();
+
+        StepVerifier.create(brandsFlux)
+                .expectNextCount(2)
+                .verifyComplete();
+    }
```

Comment:
Add one more test case that checks for no duplicate brand should be there. 
e.g
testGetAllBrands_NoDuplicateBrandIsReturned {
...
}